x-pack/plugins/security_solution/public/common/components/navigation/solution_grouped_nav/solution_grouped_nav.styles.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiThemeComputed } from '@elastic/eui';
+
+const primaryBackground = 'rgba(0, 119, 204, 0.1)';
+
+export const solutionGroupedNavStyles = (euiTheme: EuiThemeComputed<{}>) => ({
+  sideNavItem: `
+    font-weight: 400;
+    &.euiListGroupItem--primary {
+      font-weight: 700;
+    }
+    .euiListGroupItem__label {
+      width: 100%;
+    }
+    &.euiListGroupItem:focus,
+    &.euiListGroupItem:focus-within,
+    &.euiListGroupItem:hover,
+    &.euiListGroupItem-isActive {
+      background-color: ${primaryBackground};
+    }
+    &.euiListGroupItem-isActive.euiListGroupItem--primary {
+      .euiListGroupItem__button {
+        color: ${euiTheme.colors.primaryText};
+        font-weight: bold;
+      }
+    }
+    .euiButtonIcon:not([class*='isDisabled']):focus,
+    .euiButtonIcon:not([class*='isDisabled']):focus-within,
+    .euiButtonIcon:not([class*='isDisabled']):hover {
```

Comment:
😬 This file is touching way too many `.eui` classes. What are trying to achieve that you can't get out of the box with EUI and/or with custom `css` passed directly to these components?

Reply:
This is for the active and selected styles:
![primary_styles](https://user-images.githubusercontent.com/17747913/166907929-150ee0ec-0431-445a-8183-9d1be7399c12.png)
The background color is not implemented by EuiListGroupItem, I had to do it in the custom css. And the button (`extraAction`) does not allow the custom `css` prop.

I added `className` to both and I use these custom classes in the styles instead of `eui` classes, is it okay?